packages/expo-background-task/src/BackgroundTask.ts

```diff
@@ -60,6 +63,17 @@ export async function registerTaskAsync(
       `Task '${taskName}' is not defined. You must define a task using TaskManager.defineTask before registering.`
     );
   }
+  if ((await ExpoBackgroundTaskModule.getStatusAsync()) === BackgroundTaskStatus.Restricted) {
+    if (!warnAboutRunningOniOSSimulator) {
+      const message =
+        Platform.OS === 'ios'
+          ? 'On iOS, expo-background-tasks can only be used on a physical device and is not supported on the simulator.'
```

Comment:
```suggestion
          ? `Background tasks are not supported on iOS simulators. Skipped registering task: ${taskName}.`
```

how about something like this? if you like it, we could update the other message ("Background tasks are not available in the current environment") to add the task name as well

Reply:
I like it - which other message are you thinking about?

Reply:
looks like you got the other msg, i meant the one right right below the first one haha. sorry i wasn't clear